packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -3497,34 +3625,36 @@ class _ExitWidgetSelectionButtonState
     milliseconds: 100,
   );
 
+  @override
+  State<_TooltipGestureDetector> createState() =>
+      _TooltipGestureDetectorState();
+}
+
+class _TooltipGestureDetectorState extends State<_TooltipGestureDetector> {
+  Timer? _tooltipVisibleTimer;
+  Timer? _tooltipHiddenTimer;
+
   @override
   Widget build(BuildContext context) {
     return Stack(
       alignment: AlignmentDirectional.topCenter,
       children: <Widget>[
-        CustomPaint(
-          painter: _ExitWidgetSelectionTooltipPainter(
-            isVisible: _tooltipVisible,
-            buttonKey: widget.buttonKey,
-          ),
-        ),
-        MouseRegion(
-          onEnter: (_) {
-            setState(() {
-              _tooltipVisibleAfter(_tooltipDelayDuration);
-            });
+        GestureDetector(
+          onLongPress: () {
+            _tooltipVisibleAfter(_TooltipGestureDetector._tooltipDelayDuration);
+            _tooltipHiddenAfter(
+              _TooltipGestureDetector._tooltipShownOnLongPressDuration +
+                  _TooltipGestureDetector._tooltipDelayDuration,
+            );
           },
-          onExit: (_) {
-            setState(() {
-              _tooltipHiddenAfter(_tooltipDelayDuration);
-            });
-          },
-          child: Listener(
-            onPointerDown: (_) {
-              _tooltipVisibleAfter(_tooltipDelayDuration);
+          child: MouseRegion(
```

Comment:
nit: we could check kIsWeb before adding the MouseRegion widget. e.g.

```dart
Widget child = GestureDetector(
  ...
  child: widget.button,
);
if (kIsWeb) {
  child = MouseRegion(
    ...
    child: child,
  );
}
```

Reply:
I think my concern with that is we also want the MouseRegion for desktop apps, and I don't think there is a nice check for that like there is for web